pubsub/streaming-analytics/PubSubToGCS.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import datetime
+import json
+import logging
+
+import apache_beam as beam
+import apache_beam.transforms.window as window
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+class GroupWindowsIntoBatches(beam.PTransform):
+    """A composite transform that enriches Pub/Sub messages by publish
+    timestamps and then groups them based on their window information.
+    """
+
+    def __init__(self, window_size):
+        # Convert minutes into seconds.
+        self.window_size = int(window_size * 60)
+
+    def expand(self, pcoll):
+        return (pcoll
+                # Assigns window info to each element in the PCollection.
+                | beam.WindowInto(window.FixedWindows(self.window_size))
+                # Transform each element by adding publish timestamp info.
+                | 'Process Pub/Sub Message' >> (beam.ParDo(AddTimestamps()))
+                # Use a dummy key to group all the elements in this window.
+                | 'Add Dummy Key' >> beam.Map(lambda elem: (None, elem))
+                | 'Groupby' >> beam.GroupByKey()
+                | 'Abandon Dummy Key' >> beam.MapTuple(lambda _, val: val))
+
+
+class AddTimestamps(beam.DoFn):
+    """A distributed processing function that acts on the elements in
+    the PCollection.
+    """
+
+    def process(self, element, publish_time=beam.DoFn.TimestampParam):
+        """Enrich each Pub/Sub message with its publish timestamp.
+
+        Args:
+            element (bytes): A Pub/Sub message.
```

Comment:
Nit. We usually don't add parameter descriptions in our python samples but instead give example inputs.